src/main/java/org/jitsi/videobridge/ratecontrol/LastNBitrateController.java

```diff
@@ -573,7 +573,7 @@ private void add(long time, long rate)
         /**
          * Removes values added before <tt>time - period</tt>.
          */
-        private void clean(long time)
+        private synchronized void clean(long time)
```

Comment:
The synchronization seems inconsistent to me because the method clean modifies both receivedRembs.size() and sum which are both read from the unsynchronized method getAverage.

On a side note, I find the implementation of the clean method to be inefficient and more complex than really necessary:
- toRemove is an unnecessary collection that takes up instance space, grows and shrinks;
- toRemove retains elements even after it has lived up its purpose;
- Because of toRemove, more auto(un)boxing happens;
- receivedRembs is searched upon removal.
I believe a solution based on an Iterator over receivedRembs.entrySet() will be more efficient and the source code will be easier to read.